### DISCUSSION TITLE
Is tracing your Business Process with OTEL possible / wanted

### DISCUSSION THREAD

**Discussion by paule96** [open-telemetry/opentelemetry-dotnet-instrumentation#3603] [Q&A] [answered]
Hi community,

I have a maybe weird question, but from my point of view, it looks like a good thing.
So usually in an application we represent with our business logik some more complex business process.

Let's look at a short example:

Imagine we programmed a bookstore. This bookstore provides the user simple functions, like looking up new books.
But also it provides a checkout/buying process.
This process, in the backend, is splitted into smaller functions like:
- add things to your cart
- checkout
  - add address
  - add payment info
  - buy
- wait for the delivery department
- wait for delivery service
- delivered

And because it's splitted, I get nice traces for each function, but not an overview of the whole process.

And my question is, if this is even possible / wanted to do something like this with OTEL.
Some challenges with these kinds of processes is that they can take a really long time to complete (we all put things for months in the shopping cart right? XD)

And currently as far as I see the `Activity` class stops the activity when it is disposed. But business processes are kinda independent of the application/process that started the process. So the stop of the activity `checkout/buying process` shouldn't be stop when it gets disposed.

**Comment by nrcventura** [comment, marked as answer]
In order to stitch these types of things together you need to be able to pass some sort of identifier that can connect things between all of those systems. Even if you are able to do something like that, you would need this telemetry sent off/stored somewhere with enough retention to store that data for that long (and still retrieve it later). A question like this is beyond the scope of the project in this repository. However, you could try to reach out to a broader audience in one of the more general OpenTelemetry slack channels for some advice. https://github.com/open-telemetry/community?tab=readme-ov-file#discussions contains information on how to access the OpenTelemetry slack channels.
